Sources/SKCore/BuildSystemManager.swift

```diff
@@ -125,6 +125,11 @@ public final class BuildSystemManager {
     self.fallbackSettingsTimeout = fallbackSettingsTimeout
     self.buildSystem?.delegate = self
   }
+
+  public func filesDidChange(_ events: [FileEvent]) {
+    buildSystem?.filesDidChange(events)
```

Comment:
This should probably be wrapped in `queue.async`. Otherwise we are doing settings queries on the build system manager queue, but triggering file changes on the main server queue, which is racy.  Alternatively we could make the SwiftPM build system itself thread-safe, but that seems like more boilerplate.  Hopefully we can switch to swift concurrency and it won't be so annoying.

Reply:
Also running the `BuildSystem.filesDidChange` code async is beneficial I think so we don’t block the queue that receives the LSP messages.